dashboard/test/models/authentication_option_test.rb

```diff
@@ -0,0 +1,57 @@
+require 'test_helper'
+
+class AuthenticationOptionTest < ActiveSupport::TestCase
+  test 'migrated user email and hashed email look at authentication_options' do
+    original_teacher_email = 'testteacher@xyz.foo'
+    new_teacher_email = 'awesometeacher@xyz.foo'
+    teacher = create(:teacher, email: original_teacher_email)
+    email_auth = create(:email_authentication_option, user: teacher, email: new_teacher_email)
+    teacher.update(primary_email: email_auth, provider: 'migrated')
```

Comment:
For some reason the assignment operator + saving doesn't seem to work here (both the local model, and the one in the database end up unchanged), but `update` does. That disturbs me, and I'll probably have to look into why. If anyone reviewing has ideas, let me know.

Reply:
Can you provide the code? Do neither the `provider` attribute, nor the `primary_email` association change?

Reply:
Is this resolved after including `provider` in the whitelisted (select) attributes?

Reply:
No, it wasn't - that whitelisting stuff had to do with regional partners and admins, not anything blocking user updates themselves.

Now I'm seeing Circle failures that indicate I actually broke something when addressing feedback, I'll try to check that behavior again after handling that.

Reply:
Ok, after fixing whatever other bugs I had, assign/save worked. :shrug: